src/lib.rs

```diff
@@ -20,22 +20,37 @@
 //! # Examples
 //!
 //! ```no_run
+//! # fn main() -> std::io::Result<()> {
 //! use std::net::SocketAddr;
 //! use socket2::{Socket, Domain, Type};
 //!
 //! // create a TCP listener bound to two addresses
-//! let socket = Socket::new(Domain::ipv6(), Type::stream(), None).unwrap();
+//! let socket = Socket::new(Domain::ipv6(), Type::stream(), None)?;
 //!
-//! socket.bind(&"[::1]:12345".parse::<SocketAddr>().unwrap().into()).unwrap();
-//! socket.set_only_v6(false);
-//! socket.listen(128).unwrap();
+//! let address: SocketAddr = "[::1]:12345".parse().unwrap();
+//! socket.bind(&address.into())?;
+//! socket.set_only_v6(false)?;
+//! socket.listen(128)?;
 //!
 //! let listener = socket.into_tcp_listener();
 //! // ...
+//! # drop(listener);
+//! # Ok(()) }
 //! ```
 
 #![doc(html_root_url = "https://docs.rs/socket2/0.3")]
-#![deny(missing_docs)]
+#![deny(
+    missing_docs,
+    missing_debug_implementations,
+    rust_2018_idioms,
+    unused_imports,
+    dead_code
```

Comment:
Could this be left as `missing_docs` and maybe missing debug implementations? The others I think are best covered by `deny(warnings)`

Reply:
`rust_2018_idioms` and `missing_debug_implementations` are covered by warnings by default. Should I turn those into warnings and remove the rest?

Reply:
I've reduced it to `#![deny(missing_docs, missing_debug_implementations, rust_2018_idioms)]` which aren't enabled by default.